setup.py

```diff
@@ -1,41 +1,58 @@
-from setuptools import setup, find_packages
+import sys
+
 import param
+from setuptools import find_packages, setup
 
 extras_require = {
     'tests': [
-        'pytest',
         'codecov',
-        'pytest-cov',
         'flake8',
+        'geopandas',
         'hypothesis',
+        'pytest-cov',
+        'pytest',
         'scipy',
         'shapely',
-        'geopandas',
     ],
     'examples': [
-        'geopandas',
-        'matplotlib',
-        'descartes',
         'datashader',
+        'descartes',
+        'geopandas',
         'holoviews',
+        'matplotlib',
     ]
 }
 
 install_requires = [
-    'pandas>=0.25',
     'dask[complete] >=2.0',
+    'fsspec',
     'numba',
-    'numpy',
-    'pyarrow>=0.15',
+    'pandas>=0.25',
     'param',
-    'fsspec',
     'retrying',
+    'snappy',
 ]
 
+# Checking for platform explicitly because
+# pyctdev does not handle dependency conditions
+# such as 'numpy<1.20;platform_system=="Darwin"'
+if sys.platform == 'darwin':
+    install_requires.extend([
+        'numpy<1.20',
+        'pyarrow>=0.15,<2',
```

Comment:
We're using spatialpandas together with pyarrow=2 on Macs and finding massive performance improvements for parquet loading (discussed in https://github.com/pyviz-topics/examples/pull/130 somewhere, I think). Why does it need pinning back to <2 here?

Reply:
Tests have been failing on MacOS with pyarrow >=2.  After your note, I double checked and tests are still failing.  In your link, I noticed you are specifying pyarrow from conda-forge, so I am trying that next.

Reply:
So that didn't work, and putting it into the setup.py was a bad idea anyway.  Tests are still failing on Mac with pyarrow >= 2.  Here is the [error](https://github.com/holoviz/spatialpandas/pull/60/checks?check_run_id=1850783674#step:11:587), and here are the [packages](https://github.com/holoviz/spatialpandas/pull/60/checks?check_run_id=1850783674#step:9:44) that were installed.  The error seems to come from dask, so I will try pinning that next.

Reply:
Got it working with `pyarrow==2` by pinning `dask<2020.12`.